static/app/components/smartSearchBar/index.tsx

```diff
@@ -283,6 +283,12 @@ type State = {
    * autocompletion for.
    */
   searchTerm: string;
+  /**
+   * The location in the query of the search term with the negation operator skipped in the start location.
+   * If null, then there should be no search term to replace.
+   */
+  searchTermLocation: {end: number; start: number} | null;
```

Comment:
Just checking since I'm still not sure I fully understand the logic in this file yet:

Can this not be derived from `cursorToken`? Just want to make sure that the state being added here is necessary.

Reply:
@malwilley So for free text, the `cursorToken` would contain the whole block of text not just the search term, for example with | being the cursor:

`device arc|h local`

The cursor token would be the whole free text `device arch local` and not just `arch`

Reply:
@jennmueng could you use `cursorToken` along with `this.cursorPosition` to derive the search term location? Seems to me like you have all the info you need there, but maybe I'm missing something important.

What I mean is if you moved [this logic](https://github.com/getsentry/sentry/pull/36986/files#diff-5c614a18181fdef067788aa39ce60a2979507f649ed81d98af67536d7da64d7eR1276-R1301) to [this location](https://github.com/getsentry/sentry/pull/36986/files#diff-5c614a18181fdef067788aa39ce60a2979507f649ed81d98af67536d7da64d7eR1474)?

Reply:
@malwilley Hmmmmm that's true, however we need the search term that's computed with that logic passed as the query into `this.generateTagAutocompleteGroup` right.

Reply:
True, if you move this logic to a computed property `get searchTermLocation()` you could use it in both places right?

Basically, if there is any way to avoid adding another state property, I'd like to go that route

Reply:
Hmm although we'd have the logic called twice yes that would be cleaner than a new state property I agree 👍

Reply:
@jennmueng in that case extract the logic into a little helper so that you can use it in two places to get the derived value.

Agree with @malwilley that this is extra state we prob don't need